File: frontend/app/types/Appointment.ts
import type { Service } from "./Service";

export type Appointment = {
  id: string;
  guest_name: string;
  guest_email: string;
  datetime: string;
  nutritionist_id: number;
  service_id: number;
  service: Service;
  status: "pending" | "accepted" | "rejected";
};

export type CreateAppointmentDTO = Pick<
  Appointment,
  "guest_name" | "guest_email" | "datetime" | "nutritionist_id" | "service_id"
>;

export type AnswerAppointmentDTO = Pick<Appointment, "id" | "status">;

File: backend/app/models/appointment.rb
class Appointment < ApplicationRecord
    belongs_to :nutritionist
    belongs_to :service

    # validates :nutritionist_id, presence: true
    validates :datetime, presence: true
    validates :guest_name, presence: true, length: { minimum: 3 }
    validates :guest_email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
    validates :status, presence: true

    enum :status, { pending: "pending", accepted: "accepted", rejected: "rejected" }, default: "pending"
end
